spec/facets/DatasetLifecycleStateDatasetFacet.json

```diff
@@ -0,0 +1,58 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://openlineage.io/spec/facets/1-0-0/DatasetLifecycleDatasetFacet.json",
+  "$defs": {
+    "DatasetLifecycleStateDatasetFacet": {
+      "allOf": [{
+        "$ref": "https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/DatasetFacet"
+      }, {
+        "type": "object",
+        "properties": {
+          "lifecycleState": {
```

Comment:
I might go with something like `operation` or `stateChange` or something to indicate this contains the actual change, rather than simply the current state.

Reply:
The problem with `stateChange` is that it does not fit well to non-table datasets like streams, but we are still aiming to provide a facet that describes changes for tables, files and streams (as discussed here: https://github.com/MarquezProject/marquez/pull/1847#issuecomment-1033474133). 

I've changed that into `lifecycleStateChange` and hope this describes better all the possible scenarios. And yeah, it is related to the actual change, not the current state.